src/App.tsx

```diff
@@ -30,7 +30,8 @@ const Content = () => {
 }
 
 const Gateway = () => {
-    const { connection } = useConnection();
+    // you need to use an endpoint that supports web socket connections: the public solana rpc endpoint does not
+    const connection = new Connection(process.env.REACT_APP_RPC_ENDPOINT || clusterApiUrl('mainnet-beta'));
```

Comment:
I think it's still correct to use useConnection here - the connection is set up in line 43. Otherwise you end up with two connections

Reply:
thanks, I re-added